js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillLogsTab.tsx

```diff
@@ -12,48 +12,62 @@ import {
 } from '../../ticks/InstigationEventLogTable';
 import {InstigationEventLogFragment} from '../../ticks/types/InstigationEventLogTable.types';
 
-const getResultsForBackfillLogsPage = (e?: BackfillLogsPageQuery) => {
-  return e?.partitionBackfillOrError.__typename === 'PartitionBackfill'
-    ? e.partitionBackfillOrError.logEvents.events
-    : [];
-};
-const getFetchStateForBackfillLogsPage = (e?: BackfillLogsPageQuery) => {
-  return e?.partitionBackfillOrError.__typename === 'PartitionBackfill'
-    ? e.partitionBackfillOrError.logEvents
-    : null;
+const getResultForBackfillLogsPage = (e: BackfillLogsPageQuery) => {
+  if (e.partitionBackfillOrError.__typename === 'PartitionBackfill') {
+    const {events, hasMore, cursor} = e.partitionBackfillOrError.logEvents;
+    return {data: events, hasMore, cursor, error: undefined};
+  } else {
+    return {data: [], hasMore: false, error: e.partitionBackfillOrError, cursor: undefined};
+  }
 };
 
 export const BackfillLogsTab = ({backfill}: {backfill: BackfillDetailsBackfillFragment}) => {
-  const {fetched: events, refreshState} = useCursorAccumulatedQuery<
+  const {
+    error,
+    fetched: events,
+    refreshState,
+  } = useCursorAccumulatedQuery<
     BackfillLogsPageQuery,
     BackfillLogsPageQueryVariables,
-    InstigationEventLogFragment
+    InstigationEventLogFragment,
+    {message: string}
   >({
     query: BACKFILL_LOGS_PAGE_QUERY,
-    variables: {backfillId: backfill.id},
-    getResultArray: getResultsForBackfillLogsPage,
-    getNextFetchState: getFetchStateForBackfillLogsPage,
+    variables: useMemo(() => ({backfillId: backfill.id}), [backfill]),
+    getResult: getResultForBackfillLogsPage,
   });
 
-  return (
-    <>
-      <div style={{position: 'absolute', right: 16, top: -32}}>
-        <QueryRefreshCountdown refreshState={refreshState} />
-      </div>
-      {events === null ? (
+  const content = () => {
+    if (error) {
+      <Box flex={{justifyContent: 'center', alignItems: 'center'}} style={{flex: 1}}>
+        <NonIdealState title="Unable to fetch logs" description={error.message} />
+      </Box>;
+    }
+    if (events === null) {
+      return (
         <Box flex={{justifyContent: 'center', alignItems: 'center'}} style={{flex: 1}}>
           <Spinner purpose="page" />
         </Box>
-      ) : events.length > 0 ? (
-        <InstigationEventLogTable events={events} />
-      ) : (
+      );
+    }
+    if (events.length === 0) {
+      return (
         <Box flex={{justifyContent: 'center', alignItems: 'center'}} style={{flex: 1}}>
           <NonIdealState
             title="No logs available"
             description="If backfill log storage is enabled, logs will appear as they are emitted by the backfill daemon."
           />
         </Box>
-      )}
+      );
+    }
+    return <InstigationEventLogTable events={events} />;
```

Comment:
Hmm,  now I wonder if instead we should make the table load the data as you scroll rather than all at once? Are these the logs just for a single backfill? If so then what we currently have is fine and is similar to run logs

Reply:
Also does this data actually need to be refreshed? Seems like it might be immutable?

Reply:
- Hmm yeah these are logs for a single backfill, I'd like to not load them all but the cursor goes from oldest => newest because the underlying storage is files on S3, so we have to load them all to get the latest

- Yeah the goal of the polling is to make another request using the same cursor as the last request to see if it returns more rows. So the data is immutable, and calling `fetch()` over and over does not re-fetch anything already fetched, it just checks again for new data.

Reply:
Ah okay neat, I didn't realize that last detail about the cursor being maintained but it makes sense!